src/Contact/ContactController.php

```diff
@@ -4,40 +4,45 @@
 
 namespace App\Contact;
 
+use Psr\Http\Message\ResponseFactoryInterface;
 use Psr\Http\Message\ResponseInterface;
 use Psr\Http\Message\ServerRequestInterface;
+use Yiisoft\Http\Header;
 use Yiisoft\Http\Method;
+use Yiisoft\Http\Status;
+use Yiisoft\Router\UrlGeneratorInterface;
 use Yiisoft\Yii\View\ViewRenderer;
 
 class ContactController
 {
     private ContactMailer $mailer;
+    private ResponseFactoryInterface $response;
+    private UrlGeneratorInterface $url;
     private ViewRenderer $viewRenderer;
 
     public function __construct(
-        ViewRenderer $viewRenderer,
-        ContactMailer $mailer
+        ContactMailer $mailer,
+        ResponseFactoryInterface $response,
+        UrlGeneratorInterface $url,
+        ViewRenderer $viewRenderer
     ) {
         $this->mailer = $mailer;
+        $this->response = $response;
+        $this->url = $url;
         $this->viewRenderer = $viewRenderer->withControllerName('contact');
     }
 
-    public function contact(ServerRequestInterface $request, ContactForm $form): ResponseInterface
+    public function contact(ContactForm $form, ServerRequestInterface $request): ResponseInterface
```

Comment:
Why switching these two in the signature?

Reply:
Alphabet